lib/crawler/data/crawl_result/html.rb

```diff
@@ -124,7 +121,53 @@ def meta_description(limit: 1024)
           Crawler::ContentEngine::Utils.limit_bytesize(description, limit)
         end
 
-        #---------------------------------------------------------------------------------------------
+        def meta_tags_elastic(limit: 512)
+          meta_elastic_class = 'elastic'
+          meta_tag_selector = "meta.#{meta_elastic_class}"
+
+          # filter by the meta_tag_selector first to only get meta tags with class='elastic'
+          extractions = {}
+          parsed_content.css(meta_tag_selector).css('meta[name][content]').each do |meta|
+            # truncate the content field of each tag we extract
+            truncated_content = Crawler::ContentEngine::Utils.limit_bytesize(
+              meta['content'],
+              limit
+            )
+            unless !validate_field_name(meta['name']) || Constants::RESERVED_FIELD_NAMES.include?(meta['name'])
+              extractions[meta['name']] = truncated_content
+            end
+          end
+          extractions
+        end
+
+        def data_attributes_from_body(limit: 512)
+          data_elastic_name = 'data-elastic-name'
+          body_embedded_tag_selector = "[#{data_elastic_name}]"
+
+          extractions = {}
+          parsed_content.css('body').css(body_embedded_tag_selector).each do |data|
+            truncated_content = Crawler::ContentEngine::Utils.limit_bytesize(
+              data.text.to_s.squish,
+              limit
+            )
+            unless !validate_field_name(data[data_elastic_name]) ||
+                   Constants::RESERVED_FIELD_NAMES.include?(data[data_elastic_name])
+              extractions[data[data_elastic_name]] = truncated_content
+            end
+          end
+          extractions
+        end
+
+        def validate_field_name(field_name)
```

Comment:
Should this function cover the `Constants::RESERVED_FIELD_NAMES` check? Instead of having to check that separately next to calls of this function?